macros/contexts/contextSignificantFigures.pl

```diff
@@ -118,15 +113,15 @@ sub new {
 				$digits =~ s/^0+//;                      # remove leading zeros
 			}
 		}
-		$n = length($digits) unless defined($n) && $n =~ /^inf$/;    # what remains are the significant digits
+		$n = length($digits);                            # what remains are the significant digits
 	}
+	my $N = $context->checkSigFigs($n);
 
-	$context->checkSigFigs($n);
-	$value = $class->format('E', $value, $n) unless $n =~ /^inf$/;
-	$self  = bless $self->SUPER::new($context, $value + 0), $class;
-	$self->sigfigs($n);
-	$self->{data} = [$value];
-	$self->{exp}  = $n == 'inf' ? 0 : (split(/E/, $value))[1] + 0;
+	$self            = bless $self->SUPER::new($context, ROUND($value, $n)), $class;
+	$value           = $class->format('E', $self->value, $n) unless $n == 'inf';
+	$self->{data}    = [$value];
+	$self->{exp}     = $n == 'inf' ? 0 : (split(/E/, $self->value))[1] + 0;
```

Comment:
Same here.  Though `$n` will work, it is more consistent to use `$N.  The only place `$n` is needed is in the `ROUND()` call at line 120.

Also, `$self->value` can just he `$value`, as these are both the same in your code, and if you take my advice about storing the perl real rather than a string, it will *have* to be `$value` here.